server/src/app/controllers/userController.ts

```diff
@@ -0,0 +1,26 @@
+import { Request, Response } from "express";
+import { createUserAvatarService } from "../../factory";
+import { IUserAvatarService } from "../../service/IUserAvatarService";
+
+
+export const generateUserAvatar = async (req: Request, res: Response) => {
+  const userAvatarService: IUserAvatarService = createUserAvatarService();
+
+  try{
+    const name = req.query.name as string;
+    if(!name || name === ""){
+      res.sendStatus(400);
```

Comment:
Uma boa prática em API rest é sempre retornar uma mensagem com os erros para que o cliente saiba o que está acontecendo e o pq está recebendo aquele erro, nesse caso poderia ser algo do tipo: 

```suggestion
      res.json({ error:  "Name property is required and was not provided" }).sendStatus(400);
```

Mais pra frente podemos criar um erro que faz um extend do Error nativo para melhor estabelecer um padrão, mas é comum de ver um objeto contendo a prop de 'error' dentro da resposta